utils/serde/src/main/java/io/apicurio/registry/utils/serde/strategy/GlobalIdStrategy.java

```diff
@@ -47,14 +47,15 @@
     /**
      * Create InputStream from schema.
      * By default we just take string bytes.
-     *
+     * 
      * @param schema the schema
      * @return schema's input stream
      */
     default InputStream toStream(T schema) {
         if (schema instanceof byte[]) {
             return new ByteArrayInputStream((byte[]) schema);
         } else {
+            // TODO Calling "toString()" here will work for Avro, but is unlikely to work for other schemas
             return new ByteArrayInputStream(schema.toString().getBytes(UTF_8));
```

Comment:
Yeah ... that's why this is an interface, and this is a default ... since so far most of the schemas's toString returns what we want ...

Note: change this to use IoUtils ...

Reply:
Right, but it's an interface to a strategy (logic) of how to get the globalId - not of "what" to get (i.e. what type of artifact).  We can certainly have subclasses of each of the strategies to handle the different formats, but that seems wrong.  I haven't thought of a better approach though.

Reply:
>> I haven't thought of a better approach though.

Exactly :-)